sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -101,6 +101,28 @@ fn item_to_ast_nodes(
 ) -> Result<Vec<AstNode>, ErrorEmitted> {
     let attributes = item_attrs_to_map(context, handler, &item.attribute_list)?;
 
+    // Walk all the cfg attributes and check if the item should be included
+    if let Some(cfg_attrs) = attributes.get(&AttributeKind::Cfg) {
+        for cfg_attr in cfg_attrs {
+            for arg in &cfg_attr.args {
+                match arg.name.as_str() {
+                    "target" => {
+                        if let Some(sway_ast::Literal::String(value)) = &arg.value {
+                            if value.parsed != "evm" {
```

Comment:
@mohammadfawaz, so here, `"evm"` shouldn't be hardcoded, but read from the [BuildConfig](https://github.com/FuelLabs/sway/blob/a0150c0c5e2ff3a6fed54a300a131f829a4b9887/sway-core/src/build_config.rs#L17-L28). Should I just put the config in [Context](https://github.com/FuelLabs/sway/blob/a0150c0c5e2ff3a6fed54a300a131f829a4b9887/sway-core/src/transform/to_parsed_lang/context.rs#L2-L14)? Maybe this isn't the right place to evaluate the `cfg` attribute? (The [test](https://github.com/FuelLabs/sway/blob/a0150c0c5e2ff3a6fed54a300a131f829a4b9887/test/src/e2e_vm_tests/test_programs/should_pass/conditional_compilation/src/main.sw) passes so it works, but maybe there's a better place?)

Reply:
Passing in `BuildConfig` to `Context` is not a bad idea I think. It's not unusual that you'll have to read flags or fields from the manifest file and pass them to the compiler in various places. [Here's](https://github.com/FuelLabs/sway/blob/1a2d323e9c26b7f65bcc81d66a5acbb4c5459a6c/sway-core/src/lib.rs#L359) an example.